### ISSUE TITLE
Fix false positives for nested property declaration that contains `tx` or `qx` in unit-no-unknown

### ISSUE THREAD

**Issue by Nittarab** [stylelint/stylelint#5434]
> Clearly describe the bug

There are false positives for  unknown unite when you use `tx` or `qx`

> Which rule, if any, is the bug related to?

`unit-no-unknown`

> What code is needed to reproduce the bug?

e.g.

```scss
// Font Sizes
$font-sizes: (
    2x: $font-size-base * 2,
    2qx: $font-size-base * 2.25,
    2hx: $font-size-base * 2.5,
    tx: $font-size-base * 2.75,
    3x: $font-size-base * 3,
    3qx: $font-size-base * 3.25,
    3hx: $font-size-base * 3.5,
    3tx: $font-size-base * 3.75,
    4x: $font-size-base * 4,
    4qx: $font-size-base * 4.25,
    4hx: $font-size-base * 4.5,
    4tx: $font-size-base * 4.75,
    5x: $font-size-base * 5,
    5qx: $font-size-base * 5.25,
    5hx: $font-size-base * 5.5,
    5tx: $font-size-base * 5.75
)
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "unit-no-unknown": "true"
  }
}
```

> Which version of stylelint are you using?

e.g. `13.13.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?
```shell
yarn run stylelint --config .github/linters/.stylelintrc.json "app/**/*.scss"  
```
> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS nested properties declaration.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
 428:5  ✖  Unexpected unknown unit "qx"   unit-no-unknown
 432:5  ✖  Unexpected unknown unit "qx"   unit-no-unknown
 434:5  ✖  Unexpected unknown unit "tx"   unit-no-unknown
 436:5  ✖  Unexpected unknown unit "qx"   unit-no-unknown
 438:5  ✖  Unexpected unknown unit "tx"   unit-no-unknown
 440:2  ✖  Unexpected unknown unit "qx"   unit-no-unknown
 442:5  ✖  Unexpected unknown unit "tx"   unit-no-unknown
```

**Comment by jeddy3** [comment]
@Nittarab Thanks for the report and for using the template.

I can [reproduce using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIAkAzCF4AIAUAdAdpkZgEYCGATluQF6YBUANAcZgIwCu+AlgC5Ztm+AJQhGIAM4BPfLzIAPBOAkSx4CPhTcA5kuAtMuEBQ4AbGBKNZ9hYkbK8AtCfOP8ER1wDW7gO74rTF4TGCFWIxJTCDAvNw8YAFsAB14pQOCOUIMiI0goijjHbnwANzJTbihHAAsYeXSQsLt1BISYWULElLSEIMbswxBYMFNKB24NR0jo2PdHKA4kirAHGEcwDgleCATHJIoIJJgKXm4LBsymnKGYEbHTyemYwoWl7hXeNf3D49Pz3oA2gMiBksrZWDZWOEQDp3BQYIEgeCoddIPgJLcOKcSmtXstVhJHL4+NV5twUChju0nGVTJlLCBgcQALpMgC+A2ZV0Gw1GFHGjyizzmEmqEBO1TI+Cq3yOJykjggOIoFEq-ywoO5RjQHRQZAS3FMCrmePeqzc+vVfUuA21GiceoNRsKBpUxW0jm07WO70cjsNCq8MCkvnFUAuYOhKC4YAe+HW5TAhWKtMqEa1IGj+FjE3jK1MSbmXAkSTIYBgVR+-O2FHTtszMbjjgq+BglE9-KgZw6c3c6Lk0soVS78JzGjryKMQakKH5bXmtz5Avjc24yXFA94E+hbS7ZD9bd4HHhFvnRfwPgg-m3zXwlqq2lVVUobcJq9K5TTvU19ZNFnL0ruo4Ja3OS7x8D0Gr9JOIAmos+KfI4DhFOuJzOGYVo-jBcFvB8awYuYsbigyUE2th8TJKkwEQMe5Y3tccx1ME+4YgaeQaCR1qRreHgpp+w40ZE+GjKK6w7G0sicVh0LvqmMoQBIfC5khThrkkG7oeY9GDKuKowNoZjtr41R8BYpZ0d+0HQrKvzGh43h+AEllkdCBG3DWewYhwUAeHcKiFA5V5OaR3HXG5REFEkXk+Y4MDmBJTjnpe17OaFgzhR5qRHAFF6OYiTLSVCkIooMsLigigJMtCGxbDssXxdSDJVZgrLIkQHLIly9ZbKq+Aer2MC+C2FUhRmXB8DlyXBVxGbIS4axdFRw1TDAaDwvlbWDOUvhkFIDLckQxUokYdTlikG0lc0TxeOY-lkCgnwFBIlqnms123ftzXXFoFBbG4FifOGjKbUQXUg6V2hwiNmBIpd1z3Y9YmtNSRjNa1KIdawYPQnNGGvQmGKBEYUS+McRizU4834yWZZrAjZO9PYpg7XtwHuq4w3k-WuOuKx7wQFEK6Dct9O1ozIDbbtn0wdd6xRIpfVTPy5axZRCrLSQq3lUTsEwMqXMy0KsQjApQEkMraytkNxR0w9DNYEzLPS9CssmwrHrm7T-3W62K1rdDjtS44CRmKczY2wbLtG3LpuK57Ks0yrmv+zrkus+7HMR2I9ayz8PBxxb3si3bYsOxLzNByHphh5z2eGzMipHPnHuF4ntuPanFfp+zay1xmudN2brfmW9WvreLafO807EFLU8gEwHIAkwzGYzzUdTNu02i8NUOuihukfT+J1Kq904e+8n2uVZtgd7YfELNSd8hnVu19w4MP1-a2WwVqj4PYyVIwZVx5YFhnDIwWwpDmBbE4SArQpRQCav-dknJV7Hw6MZUybcijojVJ3J299Bg1W2LsGy8pT5LRtn7K+oCmS3yng-cGT8X4XXARLEu6xNgkL2AcOUqRCHHUzNwX6iUAa-2BpdABgjgHQzAZdewHC4EJQEVCIBuDYBsz6pnC+Rs-6SJQZ1DMvJ7jKXIIrbBosdatn1nXaExjqymKlB6bBl8QFbS7gwnki4TGCgbnzdiQsfbt3tu4p2wdQ7cHPhVIx3iHG+OeP4gWkwLEl3wUHDOvcs4xLuHE+MstEmC2AiPahbijDWJXvWexy4pjRwyVE4OCgFw5OXJxQQlTYnVNlsxQ0QECnjgnh4whRgqlNgCUXKhliBlhKrjXLJ7TmmjKSfGFJHcpnpJ7lEoZNwFnKTGS4sei9yli2yUuJsi11ZUNcbIuh5cCG2KKo-EAp0YDnTfmw0WTTTm5hUVGYRX8xFAzRgdZoMjWHyPYYjJRKN7nguKIpDRdSNa6IkSVdGUJMYsgzFmMceZxL7itsXVZZdJ7hOrpEvu9ZsWLPgUUr2kziUeM0dobR0TKWNl2Xi2lScDlWL1hUmCVLlIJEaecqJnEAAMWL2WTDvPOFYhNxbL2OWy7MTZSzwlkLUDEb5hZULhXgtZrMZnkrmQK6V8Z1XUi1RYLlax9WwF5cqJlLKtmCsmJgz4KyQn0K2cUWAshlyBAAEwZnMH1HejgaDHA8OePg2kjDCvnqK4arSMy7m4PuSkDhjxrD2cU+loSpZbPTZmw8OaxKFP2SncWRzi0VgzQebNJ5ZW5rIAqsuSq617kbUeE8lrNUWBtdg+1hy+XKpgiWnt5b+R9TWFWBw4pbVIVSYazxCb62lqbWsGdXpG7HAXU9YpVy0l3xhYMSdABHTIFALl-SUfi3VvsC30NJbM1sXaG1XuOLe2BnKvWl0Ld3LRmT31nvXd2r9N7w53r-UenlNax1bMYgLIkJlPUj3jbBDwrp3axWlIqFA1FaKW0fdDQqDEOAJE1gUcwZAuyKyjQcE9a7YKUeo4UbpLYPSMYUphshVF5WL07WB4weNk2XPg7Qm+tzK4RM2Wew6jznmvKk+-bUfzRE-0Bf-YF1xQVvPBR8qFsg9GooMVjDMGVF0OGCNwEgWI3rKyDLwQkw71GjpsZZuK7lrO8Fs-ZxC86PL-sdfy1y3mIoqX8w5vd1ZF1VpoTpRDImrMFDgSQYoB6l3Ppk6erzhEPLpcy8FuD1aGV3Pyz5gosAJAASgFKX9VHiuLt7MkjDqVKuRYqDBmlBKJkrvK0WlLEWPLdca8Kpdx6EOee6iNxdiaKEXO-oESVs2CuLqijAbyvkRKEkEzrYTnWPKbe23LNthJ+070Ha54pI7QvjvC+tyK0V4gNQ6PtxVV4wvNFS55LbMVvMJQrTKiAcg4w6x8hwISWzftZVbe2wYh36zjVgW2oTX2HvNFpJkRw05QwUCqB9jtGOtnY8yT1ibfWn0DcA4SY18mMxk+g+Nli+aacvsRaa6ETOxtIwmwl0pusZswR59wP6C3xPLd6OKgwHU2QgDZEAA) with:

```scss
$foo: (
    bar: baz *,
    1unit: 1,
)
```

The bug seems specific to the `*` and `,` characters, and causes `1unit` to be parsed as a value.

As PostCSS [parses the whole construct as a declaration](https://astexplorer.net/#/gist/8627f1ac6174a8819d39317b7f18b5f6/09b724e33e2e3e2c76d324c58ef874c956af13aa), I suggest we use the [`isStandardSyntaxDeclaration`](https://github.com/stylelint/stylelint/blob/v14/lib/utils/isStandardSyntaxDeclaration.js) ([around line 119](https://github.com/stylelint/stylelint/blob/v14/lib/rules/unit-no-unknown/index.js#L119)) to ignore the construct. We may also need to update the conditionals in that util to ignore "declarations" whose value start with a `(` and ends with a `)`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

Please branch off `v14` and not `master`.

**Comment by Nittarab** [comment]
Thank you for the accurate answer, I don't have so much time during these days, but I'll try to fix it in the next few months

**Comment by mattxwang** [comment]
Just poking around issues, was this closed by #5500? Seems like it, but want to make sure I'm not missing something.

**Comment by ybiquitous** [comment]
@mattxwang Thank you for telling the issue us. We can close it!
